Code/Tools/ProjectManager/Source/ProjectExportController.cpp

```diff
@@ -0,0 +1,150 @@
+        {
+            projectButton->SetProjectButtonAction(tr("Cancel"), [this] { HandleCancel(); });
+            AZ::Outcome<QString, QString> logFilePathQuery = m_worker->GetLogFilePath();
+            if (logFilePathQuery.IsSuccess())
+            {
+                projectButton->SetBuildLogsLink(QUrl::fromLocalFile(logFilePathQuery.GetValue()));
+            }
+            projectButton->SetState(ProjectButtonState::Exporting);
+
+            if (!m_lastLine.isEmpty())
+            {
+                UpdateUIProgress(m_lastLine);
+            }
+        }
+    }
+
+    const ProjectInfo& ProjectExportController::GetProjectInfo() const
+    {
+        return m_projectInfo;
+    }
+
+    void ProjectExportController::UpdateUIProgress(const QString& lastLine)
+    {
+        m_lastLine = lastLine.left(s_maxDisplayedBuiltOutputChars);
+
+        if (m_projectButton)
+        {
+            m_projectButton->SetContextualText(m_lastLine);
+        }
+    }
+
+    void ProjectExportController::HandleResults(const QString& result)
+    {
+        bool success = true;
+        if (!result.isEmpty())
+        {
+            AZ::Outcome<QString, QString> logFilePathQuery = m_worker->GetLogFilePath();
+            if (result.contains(tr("log")))
+            {
+                QMessageBox::StandardButton openLog = QMessageBox::critical(
+                    m_parent,
+                    tr(LauncherExportFailedMessage),
+                    result + tr("\n\nWould you like to view log?"),
+                    QMessageBox::No | QMessageBox::Yes);
+
+                if (openLog == QMessageBox::Yes)
+                {
+                    // Open application assigned to this file type
+                    if (!logFilePathQuery.IsSuccess())
+                    {
+                        qDebug() << "Failed to retrieve desired log file path" << "\n";
+                    }
+                    else if (!QDesktopServices::openUrl(QUrl::fromLocalFile(logFilePathQuery.GetValue())))
+                    {
+                        qDebug() << "QDesktopServices::openUrl failed to open " << m_projectInfo.m_logUrl.toString() << "\n";
+                    }
+                }
+                
+                if (logFilePathQuery.IsSuccess())
+                {
+                    m_projectInfo.m_logUrl = QUrl::fromLocalFile(logFilePathQuery.GetValue());
+                }
+            }
+            else
+            {
+                if (logFilePathQuery.IsSuccess())
+                {
+                    QMessageBox::critical(m_parent,
+                                      tr("%1\nYou can check the logs in the following directory:\n%2")
+                                        .arg(LauncherExportFailedMessage)
+                                        .arg(logFilePathQuery.GetValue()),
+                                      result);
+                }
+                else
+                {
+                    QMessageBox::critical(m_parent,
```

Comment:
Any reason logs would fail? Can we give users guidance to debug what happened?

Reply:
for this branch, the main issue is that the function `GetLogFilePath` failed to run. This means that for some reason we were not able to create the necessary folders to hold the error log and cmake logs. In practice I doubt we will run into this error, but I can make it clearer what the exact error was.

Reply:
`GetLogFilePath` also outputs its exact failure cases, so they would show up in the stack trace